Src/Newtonsoft.Json/Converters/XmlNodeConverter.cs

```diff
@@ -1177,6 +1177,11 @@ private bool IsArray(IXmlNode node)
 
         private void SerializeGroupedNodes(JsonWriter writer, IXmlNode node, XmlNamespaceManager manager, bool writePropertyName)
         {
+            if (node.ChildNodes.Count == 0)
```

Comment:
I ran this through a memory profiler and a collection is still allocated even though it will always be empty. Using XmlNode.HasChildNodes avoids allocating the ChildNodes collection. There will be other places this could be optimized.

To add this to the wrapper implementation for XNode you can test whether LastNode is null.